File: twoSum.cpp
class Solution { // 该方法即是每读到一个数就计算target减去该数的值，判断减后的值是否已经出现过
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int length = nums.size();
        vector<int> res;
        unordered_map<int,int> hash; // 由于unordered_map比map快所以选择无序哈希表
        int i;int j;
        for (i = 0;i<length;++i){
            int another = target - nums[i];
            if(hash.count(another)){
                res  = vector<int>({hash[another],i});
                return res;
            }
            hash[nums[i]] = i; // 初始刚无序哈希表是空的，然后再遍历的过程中逐渐填满
        }
        return res;
    }
};

File: twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        '''
        以下是我自己写的稚嫩的代码
        res = []
        for es_1 in nums:
            if target - es_1 in nums[nums.index(es_1)+1:]:
                res.append(nums.index(es_1))
                res.append(nums[nums.index(es_1)+1:].index(target-es_1)+nums.index(es_1)+1)
                return res
        return res
        '''
        hashmap = {}
        for index,num in enumerate(nums):
            another_num = target - num
            if another_num in hashmap:
                return [hashmap[another_num],index]
            hashmap[num] = index;
        return None
